File: app/src/main/java/networkonmainthreadexception/tipfortrip/ViewPagerAdapter.java
package networkonmainthreadexception.tipfortrip;


import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;

import java.util.List;

class ViewPagerAdapter extends FragmentPagerAdapter {

    private final List<Fragment> fragments;
    private final List<String> titles;

    ViewPagerAdapter(FragmentManager manager, List<Fragment> fragments, List<String> titles) {
        super(manager);
        if (fragments.size() != titles.size()) {
            throw new IllegalArgumentException();
        }
        this.fragments = fragments;
        this.titles = titles;
    }

    @Override
    public Fragment getItem(int position) {
        return fragments.get(position);
    }

    @Override
    public int getCount() {
        return fragments.size();
    }

    @Override
    public CharSequence getPageTitle(int position) {
        return titles.get(position);
    }

}

File: app/src/main/java/networkonmainthreadexception/tipfortrip/ViewPagerAdapter.kt
package networkonmainthreadexception.tipfortrip


import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter

internal class ViewPagerAdapter(
    manager: FragmentManager?,
    private vararg val items: Pair<String, Fragment>
) : FragmentPagerAdapter(manager) {

    override fun getItem(position: Int) = items[position].second

    override fun getCount() = items.size

    override fun getPageTitle(position: Int) = items[position].first
}

